src/CreatePhotosSlideshow/CreatePhotosSlideshow.tsx

```diff
@@ -96,41 +127,45 @@ const CreatePhotosSlideshow = (): JSX.Element => {
       setProgress(parseFloat((ratio * 100).toFixed(1)));
     });
 
-    for (const index of orderedItems) {
-      ffmpeg.FS(
-        'writeFile',
-        getCleanName(files[index]),
-        new Uint8Array(correctedFileArrayBuffer[index])
-      );
-    }
+    const inputPaths: Array<string> = [];
+    orderedItems.forEach(async (itemPosition, index) => {
+      const fileName = getCleanName(files[itemPosition].name);
+      inputPaths.push(`file ${fileName}\nduration 2`);
+      ffmpeg.FS('writeFile', fileName, await fetchFile(files[itemPosition]));
 
-    const ffconcatPath = 'slides.ffconcat';
-    const concatFileLines = ['ffconcat version 1.0']
-      .concat(
-        orderedItems.map((index) => {
-          const slideDuration = 1;
-          return `file ${getCleanName(
-            files[index]
-          )}\nduration ${slideDuration}`;
-        })
-      )
-      .join('\n');
+      // add the last file into the txt this is a quricky bug in ffmpeg
+      const isLastFile = index === orderedItems.length - 1;
+      if (isLastFile) {
+        inputPaths.push(`file ${fileName}`);
+      }
+    });
 
     ffmpeg.FS(
       'writeFile',
-      ffconcatPath,
-      Uint8Array.from(new TextEncoder().encode(concatFileLines))
+      'out.txt',
+      Uint8Array.from(new TextEncoder().encode(inputPaths.join('\n')))
     );
 
-    const slideShowWithPaddingAndBlurredBG = `-safe 0 -i ${ffconcatPath} -filter_complex format=yuv420p,rotate=2*PI,split[in_1][in_2];[in_1]scale='if(gt(a,1024/576),1000,-1)':'if(gt(a,1024/576),-1,526)':eval=frame[scaled_video];[scaled_video]format=rgba,pad=iw+50:ih+50:(ow-iw)/2:(oh-ih)/2:color=#00000000[padded_video];[in_2]scale=1024:576:force_original_aspect_ratio=increase,crop=1024:576,boxblur=20[bg];[bg][padded_video]overlay=(W-w-10)/2:(H-h-10)/2[out] -c:v libx264 -aspect 1024/576 -map [out] out.mp4 -y`;
-
-    await ffmpeg.run(...slideShowWithPaddingAndBlurredBG.split(' '));
+    const ffmpegRenderCommand = [...SLIDESHOW_COMMAND];
+    if (config.isBlured) {
+      ffmpegRenderCommand.push('-filter_complex');
+      ffmpegRenderCommand.push(`split [copy][original]; \ 
+        [copy] scale=${config.res}:force_original_aspect_ratio=increase,boxblur=20,crop=${config.res}[blured]; \
+        [original] scale=${config.res}:force_original_aspect_ratio=decrease[scaled_original]; \ 
+        [blured][scaled_original]overlay=(main_w-overlay_w)/2:(main_h-overlay_h)/2`);
+    } else {
+      ffmpegRenderCommand.push('-vf');
+      ffmpegRenderCommand.push(`
+        scale=${config.res}:force_original_aspect_ratio=decrease,pad=${config.res}:(ow-iw)/2:(oh-ih)/2:${config.background}`);
```

Comment:
It looks like the commands has changed from last one, I see aspect ratio decrease instead of increase. I think it looked better before with the photo centered rather than stretched vertically. But that can be done with a user control in the future.

Reply:
Yes i re-wrote the script to make the pipeline easier and more readable to adjust the photos and add more effects like ken burns, zoom fades etc .. 

Decrease here just mean don't scale (with respect to aspect ratio) the photo to fit the the thresholded w*h. If you notice increase is still maintained in blur images to take up all the space in the background.

We can add an option to let the user play around with it to make it fit accordingly with margins around the image just like the previous results

Reply:
Here's an example of the current generated slideshows
https://user-images.githubusercontent.com/77180905/207408555-ba0f5436-0112-4ecf-8c09-e46ce1cbaba3.mp4